__tests__/app.test.js

```diff
@@ -235,14 +243,24 @@ describe("POST /api/articles/:article_id/comments", () => {
 				expect(body.msg).toBe("Bad request");
 			});
 	});
-	test("404: should return not found when provided article_id is valid but non-existent", () => {
+	test("400: should return bad request when provided article_id is non-existent", () => {
 		const newComment = { username: "rogersup", body: "Testing!" };
 		return request(app)
 			.post("/api/articles/2000/comments")
 			.send(newComment)
-			.expect(404)
+			.expect(400)
```

Comment:
A valid but non-existant article id should result in a 404, rather than a 400 🔍
You may find handing this endpoint a valid but non-existant article_id like this triggers a PSQL error - think about how we can handle this error and the resulting PSQL error code to send the correct 404 error response 🤔

Reply:
I'd argue that trying to make a comment on a non-existent article would be a bad request? I think it makes sense to have a 404 for a get request here because it's reasonable to assume the user doesn't know if the article exists yet while searching for it but by the time they're trying to make a post/put/patch to it they should know that the article exists. My previous code accounted for this with a checkExists function but I was advised to remove this. Is there generally a best practice here or is it a matter of preference?